src/game/PowerGrid.cxx

```diff
@@ -1,28 +1,33 @@
 #include "PowerGrid.hxx"
 
-void PowerGrid::addPowerNode(PowerNode powerNode)
+PowerGrid::PowerGrid(PowerNode powerNode) : m_powerNodes{powerNode} {}
+
+void mergePowerGrids(PowerGrid &mainGrid, PowerGrid &toBeMerged)
 {
-  m_powerNodes.push_back(powerNode);
-}
+  mainGrid.m_powerNodes.insert(mainGrid.m_powerNodes.end(), toBeMerged.m_powerNodes.begin(), toBeMerged.m_powerNodes.end());
+  mainGrid.m_powerLevel = toBeMerged.m_powerLevel;
+};
+
+void PowerGrid::addPowerNode(PowerNode powerNode) { m_powerNodes.push_back(powerNode); }
```

Comment:
I think we need to check if it's already inside the vector or use set otherwise

Reply:
I'm not sure yet if I need duplicated entries (power nodes + conductive nodes like zone or buildings)